src/Exceptions/Handler.php

```diff
@@ -11,7 +11,7 @@
 use Illuminate\Routing\Router;
 use Illuminate\Validation\ValidationException;
 use InvalidArgumentException;
-use Psr\Http\Client\RequestExceptionInterface;
+use Symfony\Component\HttpFoundation\Exception\RequestExceptionInterface;
```

Comment:
FYI - https://github.com/MyParcelCOM/integration-commons/pull/22 added this. If we need this, then we also need to require `"psr/http-client": "^1.0"` in our composer.

Reply:
Why PSR if this class comes from Symfony? (included with laravel)

Reply:
@M4tini @yoan-myparcel I was wondering why my code broke after using the new `Handler` again. The issue here is that `Symfony\Component\HttpFoundation\Exception\RequestExceptionInterface` is not thrown by Guzzle on a client error. `ClientException` is thrown, which does also implement `Psr\Http\Client\RequestExceptionInterface` somewhere which is why the code worked with that change.

But I guess I can catch `Psr\Http\Message\RequestInterface` and re-throw as `Symfony\Component\HttpFoundation\Exception\RequestExceptionInterface` in my integration if that would be preferred? I'll change it like that for now then.

Reply:
I think in that case we should revert to the original handler with the PSR request exception interface. I actually do not understand why we need to catch symfony's RequestExceptionInterface in the exception handler 🤔

Reply:
@synio-wesley @M4tini  perhaps we should just handle both. I am getting a bit lost with either cases though 😃

Reply:
My only preference would be that the exception handler of the commons package does not need to be extended in every marketplace integration.

I guess that means adding `"psr/http-client": "^1.0"` to the commons dependency and catching the PSR.

Or we remove these three lines from the handler, but I guess @synio-wesley you needed that for a proper response?
```
        if ($e instanceof RequestExceptionInterface) {
            $status = (int) $e->getCode();
        }
```

Reply:
> Or we remove these three lines from the handler, but I guess @synio-wesley you needed that for a proper response?

@M4tini Yes, I need that to set a proper status code when Guzzle would generate a client exception. But right now I'm just doing this:

```
        try {
            return $this->guzzle->request('GET', $uri, $options + [
                'headers' => $headers,
            ]);
        } catch (RequestExceptionInterface $e) {
            throw new HttpException($e->getCode(), $e->getMessage(), $e); // \Symfony\Component\HttpKernel\Exception\HttpException
        }
```

That way it will get caught as `Symfony\Component\HttpFoundation\Exception\RequestExceptionInterface`

Btw, just a note but `psr/http-client` is a dependency of `guzzlehttp/guzzle`

Reply:
So for me it's also fine to keep it like it is now. It will work like that without having to extend or change the exception handler.